tests/joins_test.go

```diff
@@ -400,3 +402,23 @@ func TestNestedJoins(t *testing.T) {
 		CheckPet(t, *user.Manager.NamedPet, *users2[idx].Manager.NamedPet)
 	}
 }
+
+func TestJoinsPreload_Issue7013(t *testing.T) {
+	manager := &User{Name: fmt.Sprintf("Manager")}
+	DB.Create(manager)
+
+	var userIDs []uint
+	for i := 0; i < 21; i++ {
+		user := &User{Name: fmt.Sprintf("User%d", i), ManagerID: &manager.ID}
+		DB.Create(user)
+		userIDs = append(userIDs, user.ID)
+	}
+
+	var entries []User
+	assert.NotPanics(t, func() {
+		assert.NoError(t,
+			DB.Debug().Preload("Manager.Team").
+				Joins("Manager.Company").
+				Find(&entries).Error)
```

Comment:
We found other issues (relation is empty ) in this PR, can we cover them with test cases?

Reply:
I don't get it, is it covered by the two additional unit tests?